src/TEntityResolver.cpp

```diff
@@ -147,14 +178,20 @@ const QHash<QString, QString> TEntityResolver::scmStandardEntites = {
         {qsl("&ordf;"), qsl("ª")},
         {qsl("&laquo;"), qsl("«")},
         {qsl("&not;"), qsl("¬")},
-        {qsl("&shy;"), qsl("­")},
+        // This was an invisible soft-hyphen. However, Mudlet can't
+        // display it, even throws a debug msg. So we do like other
+        // clients, just use a std hyphen:
+        {qsl("&shy;"), QChar(0x2010)},
```

Comment:
The think about soft-hyphens is that ***they are not supposed to be visible unless the word they are in is around the end of the line*** and then they say "okay it is alright to split the word where I am and then draw me at the end of the line before the split". Mudlet does not (yet) have that capability as far as I recall....

Reply:
Yes, I totally agree about the soft-hyphen. Now, before this patch, it was not properly inserted into the mudline, so it just made a ? character. With the patch, it goes to the mudline, however, as you say, mudlet can't handle it and actually throws out debug and warning messages. In the meantime, or until there is something in Mudlet being able to handle it, I thought it might be sensible to just print a hyphen. This is what all other MXP clients do and it seemed more sensible then having the display code spit out warnings about it all over.

Reply:
Ah, yeah I guess it ought to be marked up with a `// TODO:` comment - but we don't do that any more...! :wink:

Reply:
We make issues instead, so we can get a single overview all the things that [need](https://github.com/Mudlet/Mudlet/issues?q=is%3Aopen+is%3Aissue+label%3Ahigh) [doing](https://github.com/Mudlet/Mudlet/issues?q=is%3Aopen+is%3Aissue+label%3Amedium+)!

Reply:
I get your point Vadim. Now, I'm not sure if you have plans to support soft-hyphens, so I didn't think in this direction.
It just felt wrong to return a character that makes code elsewhere break (or at least give a warning). Now the soft-hyphen &#xAD; can be input with UTF-8 encoding too.. So, well, maybe something should be done about it, maybe even an 'explicit' ignore.
While thinking about your comment, I first wondered why to return &#xAD; and have it inserted into mMudline (in TBuffer) when a special (not-yet existing) function needs to be called for it anyway. But then, I just realised, if it is send in in UTF-8 encoding it will endup exactly there as well, so, if you want to support it, it has to be dealt with in mMudline.

I now see, TBuffer.cpp:2256 already states there is an issue regarding to soft-hyphen. BTW, I also noticed what looks like
a glitch with line-wrapping under special conditions, maybe in combination with utf-8 encoding, but I need to double check and see if it can be reproduced reliably.

So, ok, I'll put it back in, but as a QChar(0x00AD) since the invisible character is really a pain to edit in the source.
That said, in the forum, Stephen already wondered if it is really save to include the Unicode chars in the source. So, if there is a wish for it, I can change this too.